test/integration-tests/IntegrationTests.Helpers/MockCollector.cs

```diff
@@ -0,0 +1,258 @@
+// <copyright file="MockCollector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using Google.Protobuf;
+using IntegrationTests.Helpers.Models;
+using Opentelemetry.Proto.Collector.Metrics.V1;
+using Xunit.Abstractions;
+
+namespace IntegrationTests.Helpers;
+
+public class MockCollector : IDisposable
+{
+    private static readonly TimeSpan DefaultSpanWaitTimeout = TimeSpan.FromSeconds(20);
+
+    private readonly ITestOutputHelper _output;
+    private readonly HttpListener _listener;
+    private readonly Thread _listenerThread;
+
+    public MockCollector(ITestOutputHelper output, int port = 4318, int retries = 5)
+    {
+        _output = output;
+
+        // try up to 5 consecutive ports before giving up
+        while (true)
+        {
+            // seems like we can't reuse a listener if it fails to start,
+            // so create a new listener each time we retry
+            var listener = new HttpListener();
+
+            try
+            {
+                listener.Start();
+                listener.Prefixes.Add($"http://*:{port}/"); // Warning: Requires admin access
```

Comment:
Is it possible to be more specific to avoid elevated privileges?
Usually, my VS/terminal is running on user permissions level.

Reply:
I think that I got that URL from the mockzipkincollector. I can change it for the mockcollector now, and if I get that working I'll get an issue submitted to do the same for the mockzipkincollector.

Reply:
It looks like I was able to get it to work with normal user permissions by switching from the wildcard to the ipaddress. I'm waiting to see if this works for the macos and linux CI pipelines too.